src/pages/preferences/NotificationsTab.tsx

```diff
@@ -17,59 +26,110 @@ export const NotificationsTab: React.FC<ProfileTabProps> = ({
 }) => {
   const [slackUsernameField, setslackUsernameField] = useState(slackUsername);
   const [notificationStatus, setNotificationStatus] = useState(notifications);
+  const dispatchBanner = useBannerDispatchContext();
+
+  const [updateUserSettings, { loading }] = useMutation<
+    UpdateUserSettingsMutation,
+    UpdateUserSettingsMutationVariables
+  >(UPDATE_USER_SETTINGS, {
+    onCompleted: () => {
+      dispatchBanner.successBanner(
+        `Your changes have successfully been saved.`
+      );
+    },
+    onError: (err) => {
+      dispatchBanner.errorBanner(
+        `Error while saving settings: '${err.message}'`
+      );
+    },
+  });
+
+  const handleSave = async (e): Promise<void> => {
+    e.preventDefault();
+    dispatchBanner.clearAllBanners();
+    try {
+      await updateUserSettings({
+        variables: {
+          userSettings: {
+            slackUsername: slackUsernameField,
+            notifications: omitTypename(notificationStatus),
+          },
+        },
+        refetchQueries: ["GetUserSettings"],
+      });
+    } catch (err) {}
+  };
 
   const hasFieldUpdates =
     slackUsername !== slackUsernameField ||
     notificationStatus !== notifications;
-  const omitTypename = (key, value) =>
-    key === "__typename" ? undefined : value;
-  const newPayload = JSON.parse(JSON.stringify(notifications), omitTypename);
+
+  const newPayload = omitTypename(notifications);
```

Comment:
I realize this code existed previously but I was wondering why we omit typename here since notifications doesn't have typeName in it's type definition.

Reply:
It does have it in the field returned from the query. graphql automatically returns it so it can identify the types. I iterate through the key/value pairs in the notification field to generate the grid so i need to remove the `__typename` field so that doesnt get generated as part of the grid.